instrumentation/log4j/log4j-appender-2.17/library/src/main/java/io/opentelemetry/instrumentation/log4j/appender/v2_17/LogEventToReplay.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.log4j.appender.v2_17;
+
+import java.util.Map;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.Marker;
+import org.apache.logging.log4j.ThreadContext;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.impl.ThrowableProxy;
+import org.apache.logging.log4j.core.time.Instant;
+import org.apache.logging.log4j.message.Message;
+import org.apache.logging.log4j.util.ReadOnlyStringMap;
+
+class LogEventToReplay implements LogEvent {
+
+  private static final long serialVersionUID = -3639807661256104630L;
+  private final LogEvent logEvent;
+
+  private final String loggerName;
+
+  private final Message message;
+  private final Level level;
+  private final Instant instant;
+  private final Throwable thrown;
+  private final Marker marker;
+  private ReadOnlyStringMap contextData;
+
+  LogEventToReplay(LogEvent logEvent, boolean captureAllContextDataAttributes) {
+    this.logEvent = logEvent;
+    this.loggerName = logEvent.getLoggerName();
+    this.message = logEvent.getMessage();
+    this.level = logEvent.getLevel();
+    this.instant = logEvent.getInstant();
+    this.thrown = logEvent.getThrown();
+    this.marker = logEvent.getMarker();
+    if (captureAllContextDataAttributes) {
+      this.contextData = logEvent.getContextData();
+    }
```

Comment:
is this condition helpful? what about case where capturing specific context data attributes?
```suggestion
    this.contextData = logEvent.getContextData();
```